code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardMapper.kt

```diff
@@ -76,11 +78,23 @@ class ContentCardMapper private constructor() {
  */
 
 fun AepUI<*, *>.getMeta(): Map<String, Any>? {
-    val template = this.getTemplate()
-    return when (template) {
+    return when (val template = this.getTemplate()) {
         is SmallImageTemplate ->
             ContentCardMapper.instance.getContentCardSchemaData(template.id)?.meta
 
         else -> null
     }
 }
+
+/**
+ * Extension function to track a custom interaction for the given [AepUI].
+ *
+ * @param interaction the custom interaction to track
+ */
+fun AepUI<*, *>.trackInteraction(interaction: String) {
+    when (val template = this.getTemplate()) {
+        is SmallImageTemplate ->
+            ContentCardMapper.instance.getContentCardSchemaData(template.id)
+                ?.track(interaction, MessagingEdgeEventType.INTERACT)
+    }
+}
```

Comment:
Given that there is now more than one extension function for AepUI needed by Messaging, can we move the extension functions for AepUI that are needed by Messaging into a separate file `ContentCardAepUIExtensions.kt` within this package itself.